text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,284 @@
+    - SchemaURL on Resource will need to be deprecated with entity-specific schema-url replacing it.  Additionally, as no Resource semantic conventions have ever stabilized, SchemaURL usage on Resource cannot be in stable components of OpenTelemetry.  Given prevalent violation of implementations around Resource merge specification, we suspect impact of this deprecation to be minimal.
+  - An OOTB "Env Variable Entity Detector" will be specified and provided vs. requiring SDK wide ENV variables for resource detection.
+- *Additionally, Resource Coordinator would be responsible for understanding Entity lifecycle events, for Entities whose lifetimes do not match or exceed the SDK's own lifetime (e.g. browser session).*
+
+#### Entity Detector
+
+The Entity detector in the SDK is responsible for detecting possible entities that could identify the SDK.  For Example, if the SDK is running in a kubernetes pod, it may provide an Entity for that pod.   SDK Entity Detectors are only required to provide identifying attributes, but may provide descriptive attributes to ensure combined Resource contains similar attributes as today's SDK.
+
+An Entity Detector would have an API similar to:
+
+```rust
+trait EntityDetector
+  pub fn detect_entities(...) -> Result<Vec<Entity>, EntityDetectionError>>
+```
+
+Where `Result` is the equivalent of error channel in the language of choice (e.g. in Go this would be `entities, err := e.detectEntities()`).
+
+#### Entity Merging and Resource
+
+The most important aspect of this design is how Entities will be merged to construct a Resource. We provide a simple algorithm for this behavior:
+
+- Construct a set of detected entities, E
+- All entity detectors are sorted by priority
+- For each entity detector
+  - For each entity detected
+    - If the entity exists in E, ignore it
+    - Otherwise, add the entity to E
+- Construct a Resource from the set E.
+
+Any implementation that achieves the same result as this algorithm is acceptable.
+
+#### Environment Variable Detector
+
+An Entity detector will be specified to allow Platform to inject entity identity information into workloads running on that platform.   For Example, the OpenTelemetry Operator could inject information about Kubernetes Deployment + Container into the environment, which SDKs can elect to interact with (through configuration of the Environment Variable Entity Detector).
+
+While details of ENV variables will be subject to change, it would look something like the following:
+
+```bash
+set OTEL_DETECTED_ENTITIES=k8s.deployment[k8s.deployment.name=my-program],k8s.pod[k8s.pod.name=my-program-2314,k8s.namespace=default]
+<run my program>
+```
+
+The minimum requirements of this entity detector are:
+
+- ENV variable can specify multiple entities (resource attribute bundles)
+- ENV variable can be easily appended or leverages by multiple participating systems, if needed.
```

Comment:
And influence priority if done right.. I.e. the difference between
```
export OTEL_DETECTED_ENTITIES=${OTEL_DETECTED_ENTITIES},${MY_PLATFORM_ENTITIES}
```
and
```
export OTEL_DETECTED_ENTITIES=${MY_PLATFORM_ENTITIES},${OTEL_DETECTED_ENTITIES},
```

Reply:
Love it, yes - This should have been a requirement all along.